webui/react/src/components/kit/internal/useResize.ts

```diff
@@ -7,44 +7,54 @@ interface ResizeInfo {
   y: number;
 }
 
-const defaultResizeInfo = {
+interface ResizeHook {
+  elementRef: RefObject<HTMLElement>;
+  ref: RefCallback<HTMLElement>;
```

Comment:
nit: it might just be me, but I personally found it a little confusing to see both `elementRef` and `ref`. they kind of both involve an element, so it wasn't initially obvious to me that one was for reading the element reference and one was for passing the element to it.

possible suggestions:
- name them `refObject` and `refCallback`?
- `getRef` and `setRef`?
- replace `elementRef` with exporting `elementRef.current` as `currentElement` or something?

defer to you if you prefer to keep this as it is.

Reply:
I believe it's a safer pass RefObject instead of the actual value of the RefObject (e.g. `elementRef.current`) because of how unmounting of components can mess with the actual value. Whereas the references will be preserved properly and will reflect accurately what snapshot holds. I did actually initially pass the element itself but it created some problems during runtime (transpiled fine).